### ISSUE TITLE
Slack: err doesn't know how to handle a response from slackbot

### ISSUE THREAD

**Issue by tetujin3d** [gbin/err#493]
a small amount of background: i have a plugin i'm writing that finds an image and posts a link to a channel. if the same link gets posted too soon, slackbot intercedes and prevents the image from unfurling (this is fine), but slackbot gives this information as a private message to the errbot user, which errbot doesn't seem to know how to handle. here's the output from the log (debug level):

```
10:28:22 DEBUG    errbot.backends.slack     Processing slack event: {u'username': \
u'slackbot', u'is_ephemeral': True, u'text': u'Pssst! I didn\u2019t unfurl \
<http://wow.zamimg.com/images/hearthstone/cards/enus/original/CS2_029.png> \
because it was already shared in this channel quite recently (within the last hour) \
and I didn\u2019t want to clutter things up.', u'ts': u'1444066101.000000', \
u'subtype': u'bot_message', u'msg': {u'do_not_unfurl_links': True}, u'type': \
u'message', u'channel': u'C04N20WTP'}
10:28:22 DEBUG    errbot.backends.slack     Handling message from a public channel
10:28:22 ERROR    errbot.backends.slack     message event handler raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/errbot/backends/slack.py", line 235, in serve_once
    event_handler(message)
  File "/usr/local/lib/python2.7/dist-packages/errbot/backends/slack.py", line 316, in _message_event_handler
    user = event[u'user']
KeyError: u'user'
```

i'm not sure how to fix this myself, so i thought i'd make a new issue here. please advise if this should go somewhere else. thanks!

**Comment by gbin** [comment]
definitely a case of a new type of message we don't handle yet.

**Comment by jamesob** [comment]
Fixed in https://github.com/gbin/err/pull/498

**Comment by tetujin3d** [comment]
awesome, thanks! i'll check it out.

**Comment by zoni** [comment]
I'm closing this one because it was fixed in #498 and went out with the recent releases, if this is still a problem on v3.1.1 then please let us know.
